app/src/main/java/com/example/peterchu/watplanner/Database/DatabaseHandler.java

```diff
@@ -493,25 +491,36 @@ private CourseComponent makeCourseComponent(Cursor cursor) {
         return ret;
     }
 
-    public List<CourseComponent> getCourseSchedule(String subject, String catalogNumber) {
+    public List<List<CourseComponent>> getCourseSchedule(String subject, String catalogNumber) {
         SQLiteDatabase db = this.getReadableDatabase();
 
-        List<CourseComponent> ret = new ArrayList<>();
+        List<List<CourseComponent>> ret = new ArrayList<>();
```

Comment:
seems like all the places invoking groupBySection make this list and pass it in as a parameter. Why not create it in the groupBySection method?